contracts/tansu/src/contract_dao.rs

```diff
@@ -777,11 +808,19 @@ pub fn public_execute(proposal: &types::Proposal) -> types::ProposalStatus {
 /// # Returns
 /// * `types::ProposalStatus` - The final status (Approved if approve > reject, Rejected if reject > approve, Cancelled if equal)
 pub fn anonymous_execute(tallies: &Vec<u128>) -> types::ProposalStatus {
-    let mut iter = tallies.iter();
-
-    let voted_approve = iter.next().unwrap();
-    let voted_reject = iter.next().unwrap();
-    let voted_abstain = iter.next().unwrap();
+    // Use get() method to access elements safely
+    let voted_approve = match tallies.get(0) {
+        Some(v) => v,
+        None => return types::ProposalStatus::Cancelled,
```

Comment:
Sorry for the nitpick but we actually don't want to mark it as cancelled if there is an issue. Otherwise that's an easy way for people to mark proposals as cancelled which is a real outcome.

Instead we want to fail. I probably need to add some tools to modify some bad votes or something like this but first I want to see if that's really needed and how people try to abuse the system.

I am also not sure if that can even happen if we get to that point in the code.

Reply:
Done